### PR TITLE
NIFI-9481 Exclude Data Transfer REST methods from DoSFilter

### PR DESCRIPTION
**Pull request by exceptionfactory**
#### Description of PR

NIFI-9481 Excludes Site to Site Data Transfer REST resources from the standard Denial-of-Service Filter configured for the Jetty Server.

S2S communication incorporates separate own timeout handling for transactions, and the DoSFilter can result in unexpected transmission failures when transferring large files using S2S with HTTP. The `DataTransferExcludedDoSFilter` is a custom extension of the standard Jetty `DoSFilter` that excludes REST resource methods having a Request URI that starts with `/nifi-api/data-transfer`.

Implementation changes include refactoring Jetty Server request filter configuration to a separate `RequestFilterProvider` interface with implementations for standard and REST API applications. Abstracting request filter configuration following this approach reduces the complexity of the `JettyServer` class and supports related unit testing improvements.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X] Have you written or updated unit tests to verify your changes?
- [X] Have you verified that the full build is successful on JDK 8?
- [X] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by gresockj — COMMENTED** [review]
Thanks for this contribution, @exceptionfactory.  I tested this out and observed that the `/nifi-api/data-transfer` requests are indeed skipping the `DoSFilter` now.  I just had one question below regarding the actual code.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/filter/RestApiRequestFilterProvider.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        filters.add(accessTokenDenialOfServiceFilter);
+
+        return filters;
+    }
+
+    private FilterHolder getAccessTokenDenialOfServiceFilter(final NiFiProperties properties) {
+        final FilterHolder filter = getDenialOfServiceFilter(properties, DoSFilter.class);
+
+        final int maxWebAccessTokenRequestsPerSecond = properties.getMaxWebAccessTokenRequestsPerSecond();
+        filter.setInitParameter("maxRequestsPerSec", Integer.toString(maxWebAccessTokenRequestsPerSecond));
```
[the remark this answers is not in the document]
Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/filter/RestApiRequestFilterProvider.java:53]: Thanks for checking @gresockj, the super class `getDenialOfServiceFilter()` method sets the `ipWhitelist` property as well as the other default property values.

[the remark this answers is not in the document]
Comment (gresockj) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/filter/RestApiRequestFilterProvider.java:53]: Ah, got it.  Thanks!

**Review by gresockj — APPROVED** [review]
Ok, I did some further testing, and have no further concerns.

Will merge!

**Pull request closed without merging**